### PR TITLE
Make GQA work

### PR DESCRIPTION
**Pull request by dirkgr**
This makes the GQA reader actually work. I have not trained an e2e model yet.

There are also some other fixes in here that were necessary to debug some issues.

### CODE REVIEW

File: allennlp/data/dataset_readers/vqav2.py [hunk 1]
```diff
@@ -455,7 +455,7 @@ class Split(NamedTuple):
 
             processed_images = self._process_image_paths(image_paths)
         else:
-            processed_images = [None for i in range(len(question_dicts))]
```
Comment (epwalsh) [inline allennlp/data/dataset_readers/vqav2.py:458]: I'm surprised the linter didn't complain about this before.

**Review by epwalsh — APPROVED** [review]
LGTM. @AkshitaB if you want to review as well I'll hold off on merging.

**Comment by AkshitaB** [comment]
> LGTM. @AkshitaB if you want to review as well I'll hold off on merging.

The CI seems to be failing.

**Comment by epwalsh** [comment]
Just made a couple fixes. Could use another review @AkshitaB

File: allennlp/data/dataset_readers/gqa.py [hunk 2]
```diff
@@ -111,73 +112,100 @@ def _read(self, split_or_filename: str):
...
+                        self.images[filename] for filename in filenames
+                    )
+                except KeyError as e:
+                    missing_filename = e.args[0]
+                    raise KeyError(
+                        missing_filename,
+                        f"We could not find an image with the name {missing_filename}. "
+                        "Because of the size of the image datasets, we don't download them automatically. "
+                        "Please download the images from"
+                        "INSERT URL HERE, "
```
Comment (dirkgr) [inline allennlp/data/dataset_readers/gqa.py:154]: Well this isn't good.

**Pull request merged**